prow/git/git.go

```diff
@@ -316,6 +334,17 @@ func (r *Repo) Config(key, value string) error {
 	return nil
 }
 
+// Remote builds a remote url from user, pasword, and a slice of path items
+func Remote(base string, user string, pass string, pathItems ...string) (*url.URL, error) {
+	newURL, err := url.Parse(base)
```

Comment:
Don't we parse this at flag load time? Why don't we store the URL object and not need to do this again later?

Reply:
That sounds like a good idea, but do we have to re-implement all these functions in this PR? This isn't going to be the last time these files are touch for a while and it might even make sense in the graphql change PR I have behind this one.

Can we get this merged and work on re-factoring and re-implementing over time? This PR is already size/L and all we did was add a flag :wink:

Reply:
I also went ahead and created an enhancement issue for this: https://github.com/kubernetes/test-infra/issues/11166

Reply:
Are we blocking approval on re-working signatures to use `*url.URL`, or can we save that for another PR?

Reply:
I would like to, yes.